src/OctoshiftCLI.Tests/AdoClientTests.cs

```diff
@@ -76,6 +76,71 @@ public async Task GetAsync_Encodes_The_Url()
                 ItExpr.IsAny<CancellationToken>());
         }
 
+        [Fact]
+        public async Task GetAsync_Retries_On_ServiceUnavailable()
+        {
+            // Arrange
+            using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.ServiceUnavailable)
+            {
+                Content = new StringContent("FIRST_RESPONSE")
+            };
+            using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("SECOND_RESPONSE")
+            };
+            var handlerMock = new Mock<HttpMessageHandler>();
+            handlerMock
+                .Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Get),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(firstHttpResponse)
+                .ReturnsAsync(secondHttpResponse);
+
+            using var httpClient = new HttpClient(handlerMock.Object);
+            var adoClient = new AdoClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, PERSONAL_ACCESS_TOKEN);
+
+            // Act
+            var returnedContent = await adoClient.GetAsync(URL);
+
+            // Assert
+            returnedContent.Should().Be("SECOND_RESPONSE");
+        }
+
+        [Fact]
+        public async Task GetAsync_Retries_On_Unauthorized()
+        {
+            // Arrange
+            using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized)
+            {
+                Content = new StringContent("FIRST_RESPONSE")
+            };
+            using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+            {
+                Content = new StringContent("SECOND_RESPONSE")
+            };
+            var handlerMock = new Mock<HttpMessageHandler>();
+            handlerMock
+                .Protected()
+                .SetupSequence<Task<HttpResponseMessage>>(
+                    "SendAsync",
+                    ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Get),
+                    ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(firstHttpResponse)
+                .ReturnsAsync(secondHttpResponse);
+
+            using var httpClient = new HttpClient(handlerMock.Object);
+            var adoClient = new AdoClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, PERSONAL_ACCESS_TOKEN);
+
+            // Act
+            var returnedContent = await adoClient.GetAsync(URL);
+
+            // Assert
+            returnedContent.Should().Be("SECOND_RESPONSE");
+        }
```

Comment:
These two tests are identical except the fact that one is testing for `Unauthorized` and the other for `ServiceUnavailable`. The good news is that `XUnit` has a feature to provide `InlineData` by using a `[Theory]` instead of a `[Fact]`, in your case you can merge these two tests into one like below:
```c#
[Theory]
[InlineData(HttpStatusCode.Unauthorized)]
[InlineData(HttpStatusCode.ServiceUnavailable]
public async Task GetAsync_Retries_On_Non_Success(HttpStatusCode httpStatusCode)
{
    // Arrange
    using var firstHttpResponse = new HttpResponseMessage(httpStatusCode)
    {
        Content = new StringContent("FIRST_RESPONSE")
    };
    ...
}
```

you can most certainly also add more non success status codes too 😄

Reply:
Oh very nice!  I love this!

Reply:
Added in 7b0d6fc6f9059a2270d480c94d1d42e068d9fc0b! :+1: